Content.Shared/Chemistry/ReagentUnit.cs

```diff
@@ -0,0 +1,184 @@
+        }
+
+        public static ReagentUnit New(string value)
+        {
+            return New(FloatFromString(value));
+        }
+
+        private static float FloatFromString(string value)
+        {
+            return float.Parse(value);
+        }
+
+        public static ReagentUnit operator +(ReagentUnit a) => a;
+
+        public static ReagentUnit operator -(ReagentUnit a) => new ReagentUnit(-a._value);
+
+        public static ReagentUnit operator +(ReagentUnit a, ReagentUnit b)
+            => new ReagentUnit(a._value + b._value);
+
+        public static ReagentUnit operator -(ReagentUnit a, ReagentUnit b)
+            => a + -b;
+
+        public static ReagentUnit operator *(ReagentUnit a, ReagentUnit b)
+        {
+            var aD = a.ShiftDown();
+            var bD = b.ShiftDown();
+            return New(aD * bD);
+        }
+
+        public static ReagentUnit operator *(ReagentUnit a, float b)
+        {
+            var aD = (float) a.ShiftDown();
+            return New(aD * b);
+        }
+
+        public static ReagentUnit operator *(ReagentUnit a, decimal b)
+        {
+            var aD = a.ShiftDown();
+            return New(aD * b);
+        }
+
+        public static ReagentUnit operator /(ReagentUnit a, ReagentUnit b)
+        {
+            if (b._value == 0)
+            {
+                throw new DivideByZeroException();
+            }
+            var aD = a.ShiftDown();
+            var bD = b.ShiftDown();
+            return New(aD / bD);
+        }
+
+        public static bool operator <=(ReagentUnit a, int b)
+        {
+            return a.ShiftDown() <= b;
+        }
+
+        public static bool operator >=(ReagentUnit a, int b)
+        {
+            return a.ShiftDown() >= b;
+        }
+
+        public static bool operator ==(ReagentUnit a, int b)
+        {
+            return a.ShiftDown() == b;
+        }
+
+        public static bool operator !=(ReagentUnit a, int b)
+        {
+            return a.ShiftDown() != b;
+        }
+
+        public static bool operator <=(ReagentUnit a, ReagentUnit b)
+        {
+            return a._value <= b._value;
+        }
+
+        public static bool operator >=(ReagentUnit a, ReagentUnit b)
+        {
+            return a._value >= b._value;
+        }
+
+        public static bool operator <(ReagentUnit a, ReagentUnit b)
+        {
+            return a._value < b._value;
+        }
+
+        public static bool operator >(ReagentUnit a, ReagentUnit b)
+        {
+            return a._value > b._value;
+        }
+
+        public override string ToString() => $"{ShiftDown()}";
+
+        public float Float()
+        {
+            return (float) ShiftDown();
+        }
+
+        public decimal Decimal()
+        {
+            return ShiftDown();
+        }
+
+        public int Int()
+        {
+            return (int) ShiftDown();
+        }
+
+        public static ReagentUnit Min(params ReagentUnit[] reagentUnits)
```

Comment:
1. you might want an overload that takes in only two plain parameters so that you don't always need to allocate when using this method.
2. Using LINQ for this also adds quite a bit of allocations and slowdown compared to just a plain loop.
2.1 <sup>LINQ has a `.Min` method</sup>

Reply:
1. done
2.1 used that instead